code/modules/power/floodlight.dm

```diff
@@ -147,8 +147,26 @@
 
 /obj/machinery/power/floodlight/Initialize(mapload)
 	. = ..()
+	RegisterSignal(src, COMSIG_OBJ_PAINTED, TYPE_PROC_REF(/obj/machinery/power/floodlight/, on_color_change))  //update light color when color changes
 	register_context()
 
+/obj/machinery/power/floodlight/proc/on_color_change(obj/machinery/power/flood_light, is_dark_color)
+	SIGNAL_HANDLER
+
+	if(setting > FLOODLIGHT_OFF)
+		update_light_state()
+
+/obj/machinery/power/floodlight/Destroy()
+	UnregisterSignal(src, COMSIG_OBJ_PAINTED)
+	. = ..()
+
+/// change light color during operation
+/obj/machinery/power/floodlight/proc/update_light_state()
+	var/light_color =  -99999 //same as NONSENSICAL_VALUE defined in set_light()
```

Comment:
why not just use `NONSENSICAL_VALUE` here? is it not a global define?

Reply:
make it a global then

Reply:
It's now a global define